test/Test/TypeRep/TypeRepMapProperty.hs

```diff
@@ -82,11 +84,17 @@ deleteInvariantSpec = it "invariantCheck (delete k b) == True" $ hedgehog $ do
     WrapTypeable (_ :: IntProxy n) <- forAll genTF
     assert $ invariantCheck (delete @n m)
 
+intersectionSpec :: Property
+intersectionSpec = it "m `intersection` (m `union` n) == m" $ hedgehog $ do
+    m <- forAll genMap
+    n <- forAll genMap
+    m `intersection` (m `union` n) === m
```

Comment:
I see that CI is failing on older GHC versions. I guess, you can either use `FpMap` newtype to compare typerep-maps, or run this property only on older GHC versions. 

https://github.com/kowainik/typerep-map/blob/73e7201861a752413f6d3f7eea19bce4b1291804/test/Test/TypeRep/TypeRepMapProperty.hs#L99-L109